tutorials/textcat_goemotions/README.md

```diff
@@ -99,34 +99,60 @@ spacy train \
-model, and add a new textcat model that you'll train, while keeping the existing
-models from the tagger and NER. This requires three changes to the config.
-
-1. Add the components to the `nlp.pipeline`.
-
-   ```ini
-   [nlp]
-   pipeline = ["tagger", "ner", "textcat"]
-   ```
-
-2. Add the "sourced" components in the `[components]` block. This tells the
-   config to build the NER and tagger components from the `en_core_web_sm`
-   config and to load their models from disk.
-
-   ```ini
-   [components]
-   tagger = {"source": "en_core_web_sm"}
-   ner = {"source": "en_core_web_sm"}
-   ```
-
-3. Specify that the tagger and NER are "frozen". This stops the weights of these
-   models from being reset, and stops the components from being updated.
-
-   ```ini
-   [training]
-   frozen_components = ["tagger", "ner"]
-   ```
+Suppose you want to keep all the functionality of the `en_core_web_sm` model
+and add the textcat model you just trained. You can do this using without
+changing your training config by using [`spacy
+assemble`](https://spacy.io/api/cli#assemble) - you'll just need to prepare a
+config describing your final pipeline.
+
+A sample config for doing this is included in
+`configs/cnn_with_pretrained.cfg`. After training the model in this project,
+you can combine it with a pretrained pipeline by running `spacy project run
+assemble`, which will save the new pipeline to `cnn_with_pretrained/`. 
+
+To make your own config to combine pipelines, the basic steps are:
+
+1. Include all the components you want in `nlp.pipeline`
+2. Add a section for each component, specifying the pipeline to source it from.
+3. If you have two components of the same type, specify the component names correctly.
+4. If necessary, specify `replace_listeners` to bundle a component with its tok2vec.
+
+You can also remove many values related to training - since you aren't running
+a training loop with `spacy assemble`, default values are fine.
+
+Let's go over the last two steps in a little more detail.
+
+By default, components have a simple default name in the pipeline, like "ner"
+or "textcat". However, if you have two copies of a component, then they need to
+have different names. If you need to change the name of a component you can do
+that by giving it a different name in `nlp.pipeline`, and specifying the name
+in the original pipeline using the `name` value in the section for the
+component.
+
+It depends on the pipeline, but often components use a Listener to just look
+for a tok2vec (or transformer) to get features from. If the tok2vec in your
+final pipeline is from the same pipeline as the component you're adding, then
+you don't have to do anything. But if a component has a different tok2vec, you
+can bundle the tok2vec with the component so that it doesn't use the wrong one.
+
+Here's an example of a component that has a different name than it had
+originally, and also uses `replace_listeners`:
```

Comment:
I would spell this out (what the old & new names are) and/or use `new_name` instead of `my_ner` to make it fool-proof.

Reply:
Thanks for the suggestion, I clarified this.